bql/semantic/expression.go

```diff
@@ -87,6 +87,80 @@ type evaluationNode struct {
 	rB string
 }
 
+type evaluationLiteralNode struct {
+	op OP
+
+	literalOnLeft bool
+
+	lS string
+	rS string
+}
+
+func (e *evaluationLiteralNode) Evaluate(r table.Row) (bool, error) {
+	// Binary evaluation
+	getLeft := func() (*table.Cell, error) {
+		var (
+			eL *table.Cell
+			ok bool
+		)
+		eL, ok = r[e.lS]
+		if !ok {
+			return nil, fmt.Errorf("comparison operations require the binding value for %q for row %q to exist", e.lS, r)
+		}
+		return eL, nil
+	}
+	getRight := func() (*table.Cell, error) {
+		var (
+			eR *table.Cell
+			ok bool
+		)
+		eR, ok = r[e.rS]
+		if !ok {
+			return nil, fmt.Errorf("comparison operations require the binding value for %q for row %q to exist", e.rS, r)
+		}
+		return eR, nil
+	}
+
+	cs := func(c *table.Cell) string {
+		if c.L != nil {
+			return strings.TrimSpace(c.L.ToComparableString())
+		}
+		return strings.TrimSpace(c.String())
+	}
+
+	leftCell, lerr := getLeft()
```

Comment:
Hi X, in the if we are actually checking that both sides have no binding, when that happens we return since we need at least one binding to evaluate the comparison.

When there is one error, it mean that is the literal we want.

Reply:
Mmh, OK.